Comp/r-sas_logistic-regr.qmd

```diff
@@ -0,0 +1,373 @@
+toc: true
+echo: true
+eval: true
+keep-hidden: true
+---
+
+# Summary
+
+## Goal
+
+Comparison of results between SAS vs R for different applications of logistic regression; where possible we try to ensure the same statistical method or algorithm is specified. However, there are some underlying differences between the algorithms in SAS vs R that cannot be (at least not easily) "tweaked". The document also provides some remarks on what parameters to look out for and what could have caused the numerical differences.
+
+## Scope
+
+::: columns
+::: {.column width="45%"}
+::: {.callout-note appearance="minimal" collapse="false"}
+## Methodologies
+
+-   Logistic regression
+-   Firth's bias-reduced logistic regression
+-   g-computation / standardization with covariate adjustment
+:::
+:::
+
+::: {.column width="55%"}
+::: {.callout-note appearance="minimal" collapse="false"}
+## Technical implementations
+
+-   SAS: `PROC LOGISTIC` (with and without firth option) and `%margins` macro\
+-   R: `stats::glm`, `logistf::logistf` and `beeca::get_marginal_effect`
+:::
+:::
+:::
+
+## Findings
+
+Below are summary of findings from a numerical comparison using example data, where possible we specify the same algorithm in R and SAS.
+
+::: {.callout-note appearance="minimal" collapse="false"}
+### Logistic regression
+
+Exact match (at 0.001 level) can be obtained using `glm` in R vs `PROC LOGISTIC` procedure (without Firth option) in SAS.
+:::
+
+::: {.callout-note appearance="minimal" collapse="false"}
+### Firth logistic regression
+
+Exact match cannot be obtained for all estimates using `logistf` vs `PROC LOGISTIC` procedure (with Firth option). More specifically:\
+- Coefficient estimate and 95% CI matched at 0.001 level;\
+- Standard error are not the same (e.g., 0.02023 for age in R vs 0.02065 in SAS);\
+- p-value is not the same (0.6288 in R for age vs 0.6348 in SAS);\
+:::
+
+::: {.callout-note appearance="minimal" collapse="false"}
+### g-computation with covariate adjustment
+
+Exact match (at 0.001 level) can be obtained using `get_marginal_effect` in R vs `%margins` macro in SAS.
+:::
+
+In the following sections, the parameterisation of logistic regression implementation (with an without Firth option) will be compared followed by numerical comparison using example data.
+
+# Prerequisites
+
+## R packages
+
+```{r, output = FALSE, message = FALSE}
+library(tidyverse)
+library(survival) # for example data
+library(logistf) # for firth regression
+library(beeca) # for covariate adjustment
+```
+
+## Data
+
+### Logistic regressions
+
+We use the `lung` dataset provided with {survival} R package. Initial data preparation involves generating a new binary outcome based on the weight change.
+
+```{r}
+lung2 <- survival::lung %>% 
+  mutate(
+    wt_grp = factor(wt.loss > 0, labels = c("weight loss", "weight gain"))
+  ) 
+glimpse(lung2)
+
+# save the dataset to be imported in SAS
+# write.csv(lung2, file = "data/lung.csv", na = ".")
+```
```

Comment:
For python, I also need to use these dataset so I saved the csv in the data folder. Maybe we can also use that version here
```suggestion
lung2 <- read.csv(../data/lung_cancer.csv)
lung2 <- lung2 %>% 
  mutate(
    wt_grp = factor(wt.loss > 0, labels = c("weight loss", "weight gain"))
  ) 
glimpse(lung2)
```